### PR TITLE
merchant favorite customer

### LINKED ISSUE #18: Merchant favorite customer
GET /api/v1/merchants/:id/favorite_customer

### PR DESCRIPTION
**Pull request by Dpalazzari**
@tmikeschu 

In this PR:

- Adds merchant favorite_customer method
- Test merchant favorite_customer at the model level
- Adds merchant favorite_customer API endpoint
- Tests merchat favorite_customer API endpoint
- All rspec tests pass
- All relevant spec harness tests pass

Closes #18 

FYI @sallymacnicholas

### CODE REVIEW

File: app/models/merchant.rb [hunk 1]
```diff
@@ -29,4 +29,14 @@ def self.with_most_items(quantity)
...
   end
+
+  def favorite_customer
+    customers
+    .select('customers.*, count(invoices.merchant_id) as customer_count')
+    .joins(invoices: [:transactions])
+    .merge(Transaction.successful)
+    .group('customers.id')
+    .order('customer_count desc')
+    .first
```
Comment (tmikeschu) [inline app/models/merchant.rb:40]: Solid! 👍

File: coverage/.last_run.json [hunk 2]
```diff
@@ -1,5 +1,5 @@
 {
   "result": {
-    "covered_percent": 99.42
+    "covered_percent": 74.9
```
Comment (tmikeschu) [inline coverage/.last_run.json:3]: @Dpalazzari Can we look into the index.html?

**Review by tmikeschu — COMMENTED** [review]
Nice work on this! I like how you mirrored the process and logic from customer#favorite_merchant.

File: coverage/.last_run.json (hunk 2 above)
Comment (Dpalazzari) [reply coverage/.last_run.json:3]: @tmikeschu  When running the full rspec suite, and viewing the simplecov index in the browser, we get an overall test coverage of 99%. Not sure what this line is doing, but it is not indicative of our testing coverage.

**Pull request merged**